crates/matrix-sdk-ui/src/timeline/pagination.rs

```diff
@@ -57,62 +71,92 @@ impl Timeline {
 
         let mut outcome = PaginationOutcome::new();
         while let Some(limit) = options.next_event_limit(outcome) {
-            let messages = self
-                .room()
-                .messages(assign!(MessagesOptions::backward(), {
-                    from: from.clone(),
-                    limit: limit.into(),
-                }))
-                .await?;
-            let chunk_len = messages.chunk.len();
-
-            let tokens = PaginationTokens { from, to: messages.end.clone() };
-            let res = match self.inner.handle_back_paginated_events(messages.chunk, tokens).await {
-                Ok(result) => Some(result),
-                Err(HandleBackPaginatedEventsError::TokenMismatch) => {
-                    info!("Start of timeline was altered since pagination was started, resetting");
+            match self.paginate_backwards_once(limit, from, &mut outcome).await? {
+                PaginateBackwardsOnceResult::Success { from: None } => {
+                    trace!("Start of timeline was reached");
+                    return Ok(ControlFlow::Break(BackPaginationStatus::TimelineStartReached));
+                }
+                PaginateBackwardsOnceResult::Success { from: Some(f) } => {
+                    from = Some(f);
+                    // continue
```

Comment:
Is that supposed to be commented out? If it's supposed to be a comment and not a `continue` instruction, perhaps we can change the comment so we don't trip on this again.

Reply:
Happy to apply whatever you feel makes most sense. We have another comment in back-pagination that says `// loop` in an empty match arm where the match is the only statement in a loop. I can also leave it out entirely, I don't really care much one way or the other.

Reply:
I don't know, something like:

```rust
// Fall-through here and we will continue the loop
``` 

Just so it doesn't look like an actual `continue` someone will uncomment accidentally, probably not important here even if they uncomment it, but I'd rather avoid the confusion.